components/src/components/modal/_modal-core.scss

```
@@ -160,6 +160,7 @@ $modals: (
   position: sticky;
   top: 0;
   background-color: var(--sdds-modal-bg);
+  z-index: 1000;
```

```suggestion
  z-index: 1;
```
I think it should be as low as possible, since we just want to fix that elements with `position:relative` overlaps the header.
If we keep it low we still allow for tooltips, popovers, and other things to overlay the header.